python/lsst/ts/epm/mib_tree_holder.py

```diff
@@ -93,58 +102,110 @@ def _add_default_elements(self) -> None:
-                        name=name,
-                        description=obj_id_match.group(1),
-                        oid=f"{parent.oid}.{obj_id_match.group(3)}",
-                        parent=parent,
-                        type=MibTreeElementType.BRANCH,
-                    )
+                if mod_id_match:
+                    self._process_mod_id(lines, line, mod_id_match)
+                elif obj_id_match:
+                    self._process_obj_id(obj_id_match)
                 elif obj_type_match:
                     self._process_obj_type(lines, line, obj_type_match)
 
+    def _process_mod_id(
+        self, lines: list[str], line: str, mod_id_match: re.Match
+    ) -> None:
+        """Utility method to process MODULE-IDENTITY entries.
+
+        If the parent exists the module gets added as a branch. If the parent
+        doesn't exist (yet), it will be added to the dict of pending modules so
+        it can be added as soon as the parent is added.
+
+        Parameters
+        ----------
+        lines : `list`[`str`]
+            The lines as read from an MIB file.
+        line : `str`
+            The current line that is processed.
+        mod_id_match : `re.Match`
+            Regex Match object used to look up the name of the module.
+        """
+        name = self._get_name_replacement(mod_id_match.group(1))
+        mib_oid_match = re.match(MIB_OID, line)
+        while not mib_oid_match:
+            self._line_num += 1
+            line = lines[self._line_num].strip()
+            mib_oid_match = re.match(MIB_OID, line)
+        parent_name = self._get_name_replacement(mib_oid_match.group(1))
+        oid = mib_oid_match.group(2)
+        parent = self._get_parent(parent_name)
+        if parent is not None:
+            self.mib_tree[name] = MibTreeElement(
+                name=name,
+                description=name,
+                oid=f"{parent.oid}.{oid}",
+                parent=parent,
+                type=MibTreeElementType.BRANCH,
+            )
+        else:
+            self.pending_modules[parent_name] = (name, oid)
+
+    def _process_obj_id(self, obj_id_match: re.Match) -> None:
+        """Utility method to process OBJECT IDENTIFIER entries.
+
+        If there is a pending MODULE-IDENTITY entry, that declares this OBJECT
+        IDENTIFIER entry as parent, the MODULE-IDENTITY entry gets added as a
+        branch as well.
+
+        Parameters
+        ----------
+        lines : `list`[`str`]
+            The lines as read from an MIB file.
+        line : `str`
+            The current line that is processed.
+        mod_id_match : `re.Match`
+            Regex Match object used to look up the name of the module.
+        """
+        name = self._get_name_replacement(obj_id_match.group(1))
+        parent_name = obj_id_match.group(2)
+        parent = self._get_parent(parent_name)
+        assert parent is not None, f"No parent found for {name=!r}"
```

Comment:
Maybe you could put `f"No parent found for {name=!r}"` to `log` for the debug purpose.

Reply:
Thanks for the suggestion. The string in the assert expression will be printed in case no parent is found so there is no need for an additional debug log statement. I'll leave the code as is.